torch/_dynamo/exc.py

```diff
@@ -251,6 +257,10 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             self.value = None
 
 
+class ObservedGeneratorExit(ObservedException):
+    pass
```

Comment:
Can you add a test that a try-catch for Exception does not catch a GeneratorExit?